### ISSUE TITLE
CREAM CE

### ISSUE THREAD

**Issue by afortiorama** [ganga-devs/ganga#1335]
We are moving to ARC so this might not matter to us anymore but on a CentOS7 UI the CREAM backend fails because it cannot find globus-url-copy anymore and cannot download stdout and stderr.

```
WARNING  exit status [127] of command globus-url-copy gsiftp://ce03.tier2.hep.manchester.ac.uk/var/cream_sandbox/northgma/CN_alessandra_forti_L_HEP_OU_Manchester_O_eScience_C_UK_vo_northgrid_ac_uk_manchester_Role_NULL_Capability_NULL_northgma009/70/CREAM706780812/OSB/__jobscript__.log file:/home/aforti/gangadir/workspace/aforti/LocalXML/7/output//__jobscript__.log
WARNING  command globus-url-copy gsiftp://ce03.tier2.hep.manchester.ac.uk/var/cream_sandbox/northgma/CN_alessandra_forti_L_HEP_OU_Manchester_O_eScience_C_UK_vo_northgrid_ac_uk_manchester_Role_NULL_Capability_NULL_northgma009/70/CREAM706780812/OSB/__jobscript__.log file:/home/aforti/gangadir/workspace/aforti/LocalXML/7/output//__jobscript__.log not found
ERROR    /bin/sh: globus-url-copy: command not found
```

**Comment by egede** [comment]
@afortiorama You have a better understanding of Grid issues. Does the statement above mean that we should retire the Cream backend?

**Comment by afortiorama** [comment]
Many sites still rely on CREAM. I don't know how many do direct job submission with ganga though. I suspect not many. If you want to keep it though you need to change the client that does the output sandbox download I would try gfal-copy.

Right now I can't use ganga to help the users still using cream because this doesn't work and if I go back to SL6 it doesn't like python 2.6 (at least not with the latest) :/

**Comment by mesmith75** [comment]
Sure we can update that to gfal - maybe we should make some of the commands configurable?

The latest ganga should work with SL6 - you just need to have python 2.7 in your PATH (we use a version from LCG on CVMFS when we install it).

**Comment by afortiorama** [comment]
I simplistically use runGanga.sh from CVMFS. I'd rather not stay there and think about the python version.

**Comment by afortiorama** [comment]
I'm using an older version now.

**Comment by afortiorama** [comment]
Hi, 

it seems the ARC-CE has the same problem. globus-url-copy is also used in this case. 

WARNING  command globus-url-copy gsiftp://ce01.tier2.hep.manchester.ac.uk:2811/jobs/GPOLDmj4lOtnCshSmpWEHpUqhcrQCoABFKDmXLERDmABFKDmj6WFUn/stdout.gz file:/home/aforti/gangadir/workspace/aforti/LocalXML/10/output//stdout.gz not found
ERROR    /bin/sh: globus-url-copy: command not found

WARNING  exit status [127] of command globus-url-copy gsiftp://ce01.tier2.hep.manchester.ac.uk:2811/jobs/GPOLDmj4lOtnCshSmpWEHpUqhcrQCoABFKDmXLERDmABFKDmj6WFUn/stderr.gz file:/home/aforti/gangadir/workspace/aforti/LocalXML/10/output//stderr.gz
WARNING  command globus-url-copy gsiftp://ce01.tier2.hep.manchester.ac.uk:2811/jobs/GPOLDmj4lOtnCshSmpWEHpUqhcrQCoABFKDmXLERDmABFKDmj6WFUn/stderr.gz file:/home/aforti/gangadir/workspace/aforti/LocalXML/10/output//stderr.gz not found

**Comment by afortiorama** [comment]
recommend arcget

**Comment by afortiorama** [comment]
Hi Mark,

I see you've added also ARC in the pull request. Could you let me know how to configure it?

One of the users still has problems after testing. He is submitting directly to ARC

```
WARNING  some files not successfully downloaded
Traceback (most recent call last):
  File "/cvmfs/ganga.cern.ch/Ganga/install/7.1.4/lib/python2.7/site-packages/ganga/GangaCore/Core/GangaThread/MTRunner/MTRunner.py", line 51, in run
    rslt = self._runner.algorithm.process(item)
  File "/cvmfs/ganga.cern.ch/Ganga/install/7.1.4/lib/python2.7/site-packages/ganga/GangaCore/Lib/LCG/GridftpSandboxCache.py", line 198, in process
    cmd = '%s %s %s' % (self.copyCommand, srcURI, destURI)
AttributeError: 'MyAlgorithm' object has no attribute 'copyCommand'
Traceback (most recent call last):
  File "/cvmfs/ganga.cern.ch/Ganga/install/7.1.4/lib/python2.7/site-packages/ganga/GangaCore/Core/GangaThread/MTRunner/MTRunner.py", line 51, in run
    rslt = self._runner.algorithm.process(item)
  File "/cvmfs/ganga.cern.ch/Ganga/install/7.1.4/lib/python2.7/site-packages/ganga/GangaCore/Lib/LCG/GridftpSandboxCache.py", line 198, in process
    cmd = '%s %s %s' % (self.copyCommand, srcURI, destURI)
AttributeError: 'MyAlgorithm' object has no attribute 'copyCommand'
Traceback (most recent call last):
  File "/cvmfs/ganga.cern.ch/Ganga/install/7.1.4/lib/python2.7/site-packages/ganga/GangaCore/Core/GangaThread/MTRunner/MTRunner.py", line 51, in run
    rslt = self._runner.algorithm.process(item)
  File "/cvmfs/ganga.cern.ch/Ganga/install/7.1.4/lib/python2.7/site-packages/ganga/GangaCore/Lib/LCG/GridftpSandboxCache.py", line 198, in process
    cmd = '%s %s %s' % (self.copyCommand, srcURI, destURI)
AttributeError: 'MyAlgorithm' object has no attribute 'copyCommand'
ERROR    fail to download job output: 42
```

**Comment by mesmith75** [comment]
@afortiorama Sorry about that, I made a mistake with that change.

I've been through it with Ian and the downloading of output with `arcget` should now work (the change in behaviour was more involved than I thought). It was fixed in #1360 so should be in the DEV version from tomorrow morning.

Let us know if you still experience issues.
